client/src/main/java/org/eclipse/hono/client/device/amqp/AmqpAdapterClientFactory.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.client.ConnectionLifecycle;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.client.MessageConsumer;
+import org.eclipse.hono.client.impl.AmqpAdapterClientFactoryImpl;
+
+import io.vertx.core.Future;
+
+/**
+ * A Vert.x based factory for creating clients for Hono's AMQP adapter.
+ */
+public interface AmqpAdapterClientFactory extends ConnectionLifecycle<HonoConnection> {
+
+    /**
+     * Creates a new factory for an existing connection.
+     *
+     * @param connection The connection to use.
+     * @param tenantId The ID of the tenant for which the connection is authenticated.
+     * @return The factory.
+     * @throws NullPointerException if any of the parameters is {@code null}
+     */
+    static AmqpAdapterClientFactory create(HonoConnection connection, final String tenantId) {
+        return new AmqpAdapterClientFactoryImpl(connection, tenantId);
+    }
+
+    /**
+     * Gets a client for sending telemetry data to Hono's AMQP protocol adapter.
+     * <p>
+     * The client returned may be either newly created or it may be an existing client for the tenant that this factory
+     * instance belongs to.
+     * <p>
+     * <b>Do not hold a reference to the returned sender.</b> For each send operation retrieve the sender from the
+     * factory to ensure that it contains a valid and open AMQP link.
+     *
+     * @return A future that will complete with the sender once the link has been established. The future will fail if
+     *         the link cannot be established, e.g. because this client is not connected.
+     * 
+     * @see "https://www.eclipse.org/hono/docs/dev/user-guide/amqp-adapter/"
+     */
+    Future<TelemetrySender> getOrCreateTelemetrySender();
+
+    /**
+     * Gets a client for sending events to Hono's AMQP protocol adapter.
+     * <p>
+     * The client returned may be either newly created or it may be an existing client for the tenant that this factory
+     * instance belongs to.
+     * <p>
+     * <b>Do not hold a reference to the returned sender.</b> For each send operation retrieve the sender from the
+     * factory to ensure that it contains a valid and open AMQP link.
+     *
+     * @return A future that will complete with the sender once the link has been established. The future will fail if
+     *         the link cannot be established, e.g. because this client is not connected.
+     *
+     * @see "https://www.eclipse.org/hono/docs/dev/user-guide/amqp-adapter/"
+     */
+    Future<EventSender> getOrCreateEventSender();
+
+    /**
+     * Creates a client for consuming commands from Hono's AMQP protocol adapter for a specific device.
+     * <p>
+     * When implementing a (protocol) gateway, this can be used to receive commands for a specific device.
+     * <p>
+     * The command passed in to the command consumer will be settled automatically if the consumer does not throw an
+     * exception.
```

Comment:
I guess `if the messageHandler does not throw an exception` is meant here. However, in that case the command message would also be settled (see https://github.com/eclipse/hono/blob/525b3aa4ccabbcfbc1c97eef53684efb5d7934da/client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java#L823). In general, the message handler is not really supposed to throw an exception, though.
Therefore I think the `if` part of the sentenced can be removed here.